test/e2e/framework/util.go

```diff
@@ -3137,7 +3137,7 @@ func WaitForDeploymentRevisionAndImage(c clientset.Interface, ns, deploymentName
 	var deployment *extensions.Deployment
 	var newRS *extensions.ReplicaSet
 	var reason string
-	err := wait.PollImmediate(Poll, 1*time.Minute, func() (bool, error) {
+	err := wait.Poll(Poll, 1*time.Minute, func() (bool, error) {
```

Comment:
To be fair, my complaint was about using the messages in the events. If we enhanced events to have API stable event codes and data in addition to messages, then using events would be preferred.